runtime/libpgmath/lib/common/mthdecls.h

```diff
@@ -34,7 +34,7 @@
 #define I_SIZE 2 /* i[] has 2 elements and 128 bits */
 #define LITTLE_SIGN_MASK  0x7fffffff
 #define BIG_SIGN_MASK 0xffffff7f
-
+typedef long double float128_t;
```

Comment:
I have added `float128.h` in both `libpgmath` and `runtime` to define a type `float128_t` which is either `__float128` or `long double`, depending on the target. The code should use `float128_t` instead of `long double`.